kafka_consumer_freshness_tracker/src/main/java/com/tesla/data/consumer/freshness/ConsumerFreshness.java

```diff
@@ -113,22 +127,73 @@ public static void main(String[] args) throws IOException, InterruptedException
   }
 
   private void setup(Map<String, Object> conf) {
-    this.burrow = new Burrow((Map<String, Object>) conf.get("burrow"));
+    setupWithBurrow(conf, new Burrow((Map<String, Object>) conf.get("burrow")));
+  }
+
+  @VisibleForTesting
+  void setupWithBurrow(Map<String, Object> conf, Burrow burrow) {
+    this.burrow = burrow;
     int workerThreadCount = (int) conf.getOrDefault("workerThreadCount", DEFAULT_WORKER_THREADS);
     this.availableWorkers = ((List<Map<String, Object>>) conf.get("clusters")).stream()
         .map(clusterConf -> {
+          // validate the cluster configuration
+          ValidationResult result = validateClusterConf(clusterConf);
+          if (!result.valid) {
+            LOG.error("configuration for cluster " + (String) clusterConf.get("name") + " is invalid", result.reason);
+            return null;
```

Comment:
dont we want to abort if validation fails?

Reply:
Good question, we certainly can if that is the preferred behaviour.

Reply:
If we don't fail, what's the point of strict?

Reply:
fixed